src/gamescope/wayland/metrics/mango_app_reader.rs

```diff
@@ -0,0 +1,44 @@
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use gamescope_wayland_client::mangoapp::{message::MangoAppMsgV1, queue::MangoAppMsgQueue};
+
+pub struct MangoAppMsgQueueReader {
+    queue: MangoAppMsgQueue,
+    rx: std::sync::mpsc::Receiver<()>,
+    tx: tokio::sync::watch::Sender<Option<MangoAppMsgV1>>,
+}
+
+impl MangoAppMsgQueueReader {
+    pub fn new(
+        rx: std::sync::mpsc::Receiver<()>,
```

Comment:
We should use a tokio receiver here. It supports for async and sync operations:

```suggestion
        rx: tokio::sync::mpsc::Receiver<()>,
```